oauth2client/dictionary_storage.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Dictionary storage for OAuth2 Credentials."""
+
+from oauth2client.client import OAuth2Credentials
+from oauth2client.locked_storage import LockedStorage
+
+
+__author__ = 'jonwayne@google.com (Jon Wayne Parrott)'
+
+
+class DictionaryStorage(LockedStorage):
```

Comment:
Absolutely, I will. The core of the issue here is that I feel like I'm currently working around `Storage`'s current and somewhat surprising design because I'm trying not to make any breaking changes to the core of `oauth2client`. This is also the driving factor behind me wanting to add `contrib`, but I need some changes to the core first before I feel comfortable doing everything in contrib.